stripe/api_requestor.py

```diff
@@ -275,8 +282,7 @@ def request_headers(self, api_key, method):
             headers["Content-Type"] = "application/x-www-form-urlencoded"
             headers.setdefault("Idempotency-Key", str(uuid.uuid4()))
 
-        if self.api_version is not None:
-            headers["Stripe-Version"] = self.api_version
+        headers["Stripe-Version"] = self.api_version
```

Comment:
See line 85. `stripe.api_version` is set by default (now because of version pinning), and so self.api_version will always be set. Pyright was complaining about this if statement being redundant, so I removed it.

Reply:
Do we support setting `stripe.api_version` to None?

Reply:
* The types prohibit it.
* Even if it is `None` here, setting `headers["Stripe-Version"]` explicitly to None ultimately has the same result, I tested
```
>>> import stripe
>>> stripe.api_key = "xyz"
>>> stripe.api_version = None
>>> a = stripe.api_requestor.APIRequestor()
>>> a.api_base = "http://localhost:4444"
>>> a.request_raw("post", "/foo", {"bar": "baz"})
```

```
$ nc -l 4444
POST /foo HTTP/1.1
Host: localhost:4444
User-Agent: Stripe/v1 PythonBindings/7.0.0
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive
X-Stripe-Client-User-Agent: ...
Authorization: Bearer xyz
Content-Type: application/x-www-form-urlencoded
Idempotency-Key: 8669fffb-835d-479d-be81-3c93bc26ea63
Content-Length: 7

bar=baz
```